crates/opencascade/src/primitives.rs

```diff
@@ -372,6 +455,74 @@ impl Face {
         Shape { inner }
     }
 
+    /// Fillets the face edges by a given radius at each vertex
+    pub fn fillet(&mut self, radius: f64) {
+        // use BRepFilletAPI_MakeFillet2d
+        let mut make_fillet = BRepFilletAPI_MakeFillet2d_ctor(&self.inner);
+
+        // add all vertices from the face
+        let face_shape = cast_face_to_shape(&self.inner);
+
+        // We use a shape map here to avoid duplicates.
+        let mut shape_map = new_indexed_map_of_shape();
+        map_shapes(face_shape, TopAbs_ShapeEnum::TopAbs_VERTEX, shape_map.pin_mut());
+
+        for i in 1..=shape_map.Extent() {
+            let vertex = TopoDS_cast_to_vertex(shape_map.FindKey(i));
+            BRepFilletAPI_MakeFillet2d_add_fillet(make_fillet.pin_mut(), vertex, radius);
+        }
+
+        make_fillet.pin_mut().Build(&Message_ProgressRange_ctor());
+
+        let result_shape = make_fillet.pin_mut().Shape();
+        // convert back to a wire with BRepTools::OuterWire
+        let result_face = TopoDS_cast_to_face(result_shape);
+
+        self.inner = TopoDS_Face_to_owned(result_face);
+    }
+
+    /// Chamfer the wire edges at each vertex by a given distance
+    ///
+    /// If distance2 is None, then the chamfer is symmetric
+    pub fn chamfer(&mut self, distance1: f64, distance2: Option<f64>) {
+        let distance2 = distance2.unwrap_or(distance1);
+
+        // add all vertices from the face
+        let face_shape = cast_face_to_shape(&self.inner);
+
+        // use BRepFilletAPI_MakeFillet2d for 2d face
+        let mut make_fillet = BRepFilletAPI_MakeFillet2d_ctor(&self.inner);
+        let mut edge_map = new_indexed_map_of_shape();
+        map_shapes(face_shape, TopAbs_ShapeEnum::TopAbs_EDGE, edge_map.pin_mut());
+
+        // chamfer at vertex of all edges
+        for i in 1..=edge_map.Extent() - 1 {
+            let edge1 = TopoDS_cast_to_edge(edge_map.FindKey(i));
+            let edge2 = TopoDS_cast_to_edge(edge_map.FindKey(i + 1));
```

Comment:
From what I understand, the `edge_map` will remove duplicates, but does it guarantee any sort of order in the map? If not, grabbing `edge_1` and `edge_2` in the map as consecutive entries doesn't necessarily mean they're connected edges.

From what I saw of the cadquery code, we need to ask opencascade to give us a mapping of `Vertex` -> (`Edge`, `Edge)`, where the two edges are connected by the given `Vertex`.